account_banking_reconciliation/account_banking_reconciliation.py

```diff
@@ -0,0 +1,1224 @@
+        account_move_line_obj = self.pool.get('account.move.line')
+        statement_line_obj = self.pool.get('bank.acc.rec.statement.line')
+        val = {
+            'value': {
+                'credit_move_line_ids': [],
+                'debit_move_line_ids': [],
+                'multi_currency': False,
+                'company_currency_id': False,
+                'account_currency_id': False,
+            }
+        }
+        if account_id:
+            last_rec = self._get_last_reconciliation(cr, uid, account_id,
+                                                     context=context)
+            if last_rec and last_rec.ending_date:
+                val['value']['exchange_date'] = (datetime.strptime(
+                    last_rec.ending_date, DEFAULT_SERVER_DATE_FORMAT,
+                ) + timedelta(days=1)).strftime(DEFAULT_SERVER_DATE_FORMAT)
+            elif ending_date:
+                # end_date - 1 month + 1 day
+                dt_ending = datetime.strptime(
+                    ending_date, DEFAULT_SERVER_DATE_FORMAT,
+                ) + timedelta(days=-1)
+                if dt_ending.month == 1:
+                    dt_ending = dt_ending.replace(year=dt_ending.year - 1,
+                                                  month=12)
+                else:
+                    dt_ending = dt_ending.replace(month=dt_ending.month - 1)
+                val['value']['exchange_date'] = dt_ending.strftime(
+                    DEFAULT_SERVER_DATE_FORMAT)
+
+            account = account_obj.browse(cr, uid, account_id, context=context)
+            acur = account.currency_id
+            ccur = account.company_id.currency_id
+            if acur and ccur and acur.id != ccur.id:
+                val['value']['multi_currency'] = True
+
+            for statement in self.browse(cr, uid, ids, context=context):
+                statement_line_ids = statement_line_obj.search(
+                    cr, uid,
+                    [('statement_id', '=', statement.id)],
+                    context=context
+                )
+                # call unlink method to reset and remove existing
+                # statement lines and mark reset field values in
+                # related move lines
+                statement_line_obj.unlink(
+                    cr, uid, statement_line_ids, context=context
+                )
+
+            # Apply filter on move lines to allow
+            # 1. credit and debit side journal items in posted state
+            #    of the selected GL account
+            # 2. Journal items which are not assigned to previous
+            #    bank statements
+            # 3. Date less than or equal to ending date provided the
+            #    'Suppress Ending Date Filter' is not checked
+            # get previous uncleared entries
+            domain = [
+                ('account_id', '=', account_id),
+                ('move_id.state', '=', 'posted'),
+                ('cleared_bank_account', '=', False)
```

Comment:
as @fanzalika pointed out, `('journal_id.type', '!=', 'situation')` should be added